Draft documentation for the metadata-driven release process (#364)
camaraproject/ReleaseManagement: documentation/SupportingDocuments/camara_release_management_documentation_proposed_scaffold.md

│   ├── README.md
│   ├── API-Readiness-Checklist.md
│   ├── API_Release_Guidelines.md
│   └── CHANGELOG_TEMPLATE.md
│
└── SupportingDocuments/               # Existing template folder (unchanged)
```

---

## Deprecation & Link-Stability Strategy

### 1. Files that currently exist at top level

For documents that currently live directly under `documentation/`:

- They **must remain reachable at their original paths** during migration
- Two acceptable patterns (choose one policy and apply consistently):

**Option A – Keep file, add deprecation banner**

- File remains where it is
- A short banner at the top states:
  - Document is deprecated
  - Canonical replacement location
  - Date / phase of deprecation

**Option B – Replace with stub redirect**

- Original file is replaced with a short Markdown file
- Content:
  - Deprecation notice
  - Link to new canonical document under `deprecated/`

This avoids breaking inbound links while making status explicit.

**Current preference (not final):** use **Option B (stubs)** for link stability and clarity, unless we find a concrete reason to keep full content in-place.

### 2. `deprecated/` folder role

> - Acts as the **canonical archive** of superseded material

Reviewer: ```suggestion
- Acts as the **archive** of superseded material
```

- Is explicitly linked from the new `README.md`
- Makes deprecation intentional and visible
- Prevents legacy documents from being silently lost

---

## Documentation Philosophy (Reminder)

- One document = one responsibility
- Metadata files are **user interfaces**, not just configuration
- Declarative intent over procedural instructions
- Automation behavior is explained, not reimplemented
- Migration clarity is more important than structural purity

---

## Next Iteration Hooks

Suggested areas for your inline comments:

- Folder naming (`release-process` vs `process`, etc.)
- Whether any documents should be merged or split
- Placement and handling of `API-Readiness-Checklist`
- Whether `deprecated/` should mirror exact filenames or use aliases
- README routing wording and granularity

(We can then iterate section-by-section.)

---

## Per-Document Outlines

> The following outlines define **scope, audience, and boundaries** for each proposed document. They are intentionally explicit to avoid overlap and scope creep.

Reviewer: ```suggestion
The following outlines define **audience, purpose and scope (what is in and out of scope)** for each proposed document. They are intentionally explicit to avoid overlap and scope creep.
```

- **Metadata** describes *declared intent* and *recorded outcomes*
- **Readiness** defines *quality and maturity gates*
- **Automation** enforces rules and produces repeatable release artifacts

These parts are intentionally separated but work together.

---

## Transitional and deprecated documents

During migration, some legacy documents are still present for link stability:

- `API-Readiness-Checklist.md`
- `API_Release_Guidelines.md`
- `CHANGELOG_TEMPLATE.md`

These documents are **deprecated** and replaced by the documentation in this folder. See `deprecated/README.md` for details.

---

## Migration status

- The metadata-driven release process is being rolled out incrementally
- Legacy documents are kept temporarily to avoid breaking links
- Over time, the documents in this folder become the authoritative reference

---

**If you are unsure where to start:**\
Start with `release-process/terminology.md`, then `release-process/overview.md`.

---

**Must NOT cover**

- Release process details
- Metadata field explanations

---

## release-process/

### `terminology.md`

**Audience:** Newcomers, codeowners, maintainers, reviewers\

> **Responsibility:** Canonical terminology and mental models for the release system

Reviewer: ```suggestion
**Responsibility:** Define terminology and models underlying the release process
```

**Outline**

- Why terminology matters (common confusions)
- Core entities and how they relate
  - Repository release vs API version
  - Release tag (`rX.Y`) vs API SemVer (`X.Y.Z`)
  - Release type vs API status
  - Snapshot / release attempt vs published release
  - `release-plan.yaml` (intent) vs `release-metadata.yaml` (record)
- Naming conventions and examples (minimal)
- “If you remember one thing…” summary

**Must NOT cover**

- Full process walkthrough
- Schema reference or field-by-field definitions

---

### `overview.md`

**Audience:** Codeowners, maintainers\
**Responsibility:** High-level mental model of the release process

**Outline**

> - Why CAMARA has a structured release process

Reviewer: In the below, I will put "background" if not directly addressing day to day release process, which is the case for this point

background

**Outline**

- What qualifies as a maintenance release
- Maintenance branches vs `main`
- Metadata implications
- When *not* to use maintenance releases

**Must NOT cover**

- CI job details
- Hotfix governance rules

---

### `migration-and-adoption.md`

**Audience:** All roles\
**Responsibility:** Explain coexistence of old and new release processes

**Outline**

- Why migration is phased
- What is authoritative today
- How legacy docs map to new docs
- Expected adoption milestones

**Must NOT cover**

- Historical release timelines

---

## metadata/

### `concepts.md`

**Audience:** Codeowners, reviewers\
**Responsibility:** Explain metadata as declarative release intent

**Outline**

- Metadata as UX, not configuration
- Planning vs generated metadata
- Source-of-truth rules
- Immutability and traceability

**Must NOT cover**

- Field-by-field schema reference

---

### `release-plan.md`

**Audience:** Codeowners\
**Responsibility:** How to use `release-plan.yaml`

**Outline**

- Purpose and ownership
- Repository-level intent
- API-level intent
- Dependencies
- Common mistakes

**Must NOT cover**

- Release creation steps
- Generated values

---

### `release-metadata.md`

> **Audience:** Release managers, tooling authors\

Reviewer: given the audience, this section seems more background ?
should add maintainers ? or rather add the creation and its usage explanation in the previous section

**Responsibility:** Explain generated `release-metadata.yaml`

**Outline**

- When and how it is generated
- Relationship to tags and snapshots
- How it is used for reporting and audit

**Must NOT cover**

- Editing instructions

---

### `validation-and-freeze.md`

**Audience:** Codeowners, maintainers\
**Responsibility:** Explain validation, blocking, and freeze behavior

**Outline**

- Validation vs enforcement
- Status compatibility
- Metadata-only vs mixed PRs
- Configuration freeze during release
- Recovery paths

**Must NOT cover**

- CI job names

---

## readiness/

### `readiness-model.md`

**Audience:** Codeowners\
**Responsibility:** Conceptual readiness model

**Outline**

> - Why readiness exists

Reviewer: ```suggestion
- What is readiness
```

- API status levels
- Repository readiness vs API readiness

**Must NOT cover**

- Checklist details

---

### `api-readiness-checklist.md`

**Audience:** Codeowners, reviewers\

> **Responsibility:** Canonical explanation of the checklist

Reviewer: ```suggestion
**Responsibility:** Explanation of the readiness checklist
```

**Outline**

- Purpose of the checklist
- How it maps to statuses

> - How it is reviewed

Reviewer: ```suggestion
- How readiness is reviewed
```

**Must NOT cover**

- Release process explanation

---

### `status-progression.md`

**Audience:** Codeowners\
**Responsibility:** Explain allowed API status transitions

**Outline**

- Draft → Alpha → RC → Public
- Promotion expectations
- Locking behavior after public release

**Must NOT cover**

- CI internals

---

## automation/

> ### `automation-overview.md`

Reviewer: more a background section 
a summary of this should be a paragraph in the overview.md ?

**Audience:** All roles\
**Responsibility:** Explain what automation does and guarantees

**Outline**

- Automation as co-reviewer (why it exists)

> - Two automation surfaces

Reviewer: ```suggestion
- Two automation objectives
```

>   - Continuous validation for PRs to `main` (planning & development)

Reviewer: ```suggestion
  - Continuous validation of PRs to `main` (planning & development)
```

  - Release automation for creating and publishing releases (snapshot-based)
- What automation owns vs humans
- Guarantees and non-goals

**Must NOT cover**

- Implementation details

---

### `triggers-and-entry-points.md`

**Audience:** Codeowners, maintainers, release managers\

> **Responsibility:** Happy-path interaction points for release automation (how humans drive a release)

Reviewer: ```suggestion
**Responsibility:** Happy-path: how to drive a release assisted by automation
`

**Outline**

- Entry points and intent vs state
- Release Issue
  - Purpose (UI + audit trail; not the source of truth)
  - Commands (what they do at a high level)
- Release PR (release preparation PR)
  - What to do in the PR (what is reviewable)
  - Who must approve (release reviewers vs release managers; repo codeowners as needed)
  - Who merges and what merge means (moves snapshot forward; not yet published)

> - Draft release publication (final human control point)

Reviewer: ```suggestion
- Draft release creation (last human control point before release)
```
